File: AndroidIntro/Test2Activitati/app/src/main/java/com/example/alexandra/test2activitati/SecondActivity.java
package com.example.alexandra.test2activitati;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class SecondActivity extends Activity {

    String editText = "";
    EditText editText2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        editText2 = (EditText) findViewById(R.id.editText2);

        String getEditText = getIntent().getStringExtra("EditText1");

        if (getEditText != null) {
            editText2.setText(getEditText);
        }
    }

    public void back(View v) {
        editText = editText2.getText().toString();
        Intent intent = new Intent(SecondActivity.this, MainActivity.class);
        intent.putExtra("EditText2", editText);
        startActivity(intent);
    }
}

File: KotlinIntro/Test2Activitati2/app/src/main/java/com/example/alexandra/test2activitati/SecondActivity.kt
package com.example.alexandra.test2activitati

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText

class SecondActivity : AppCompatActivity() {

    private var editText = ""
    private lateinit var editText2: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        editText2 = findViewById<View>(R.id.editText2) as EditText

        val getEditText = intent.getStringExtra("EditText1")

        if (getEditText != null) {
            editText2.setText(getEditText)
        }
    }

    fun back(v: View) {
        editText = editText2.text.toString()
        val intent = Intent(this@SecondActivity, MainActivity::class.java)
        intent.putExtra("EditText2", editText)
        startActivity(intent)
    }
}
